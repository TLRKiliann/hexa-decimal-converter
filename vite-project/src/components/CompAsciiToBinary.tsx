import React, { useState } from 'react'
import LabelInputComponent from './LabelInputComponent.tsx'
import ResultComponent from './ResultComponent.tsx'
import ButtonComponent from './ButtonComponent.tsx'
import { convertResult } from '../functions/asciiToBinary.tsx'
import { converterTarget } from '../functions/converterTarget.tsx'

function CompAsciiToBinary(): JSX.Element {

    const [value, setValue] = useState<string>("");
    const [result, setResult] = useState<string | undefined>("");

    const nameValue = "ASCII";

    const handleEvent = (event: React.ChangeEvent<HTMLInputElement>) => {
        let eventVal = event.target.value;
        converterTarget({eventVal, setValue});
    }

    const handleClick = () => {
        convertResult({value, setValue, setResult});
    }

    return (
        <>
            <div className="labelinput--box">
                <LabelInputComponent 
                    value={value}
                    nameValue={nameValue}
                    converterTarget={(event)=>handleEvent(event)}
                >{nameValue}</LabelInputComponent>
            </div>

            <ButtonComponent convertResult={handleClick} />
            
            <div className="box--result">
                <ResultComponent 
                    result={result}
                />
            </div>
        </>
    )
}

export default CompAsciiToBinary