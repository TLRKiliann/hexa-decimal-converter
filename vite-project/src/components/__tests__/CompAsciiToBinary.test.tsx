import { test, expect } from "vitest"
//import { fireEvent, screen, render } from '@testing-library/react'
import { create } from "react-test-renderer"
import "@testing-library/jest-dom"
import CompAsciiToBinary from '../CompAsciiToBinary';
//import LabelInputComponent from "../LabelInputComponent"

test('Snapshat of CompAsciiToBinary', () => {
    const tree = create(<CompAsciiToBinary />)
    expect(tree.toJSON()).toMatchSnapshot()
});

/*
describe("testing children FirstComp", () => {
    test('testing function handleReverse', () => {
        const beverage = "tested";
        const convertResult = vi.fn(beverage => beverage)
        convertResult(beverage)
        render(
            <LabelInputComponent
            value={undefined} nameValue={""}>
            <button data-testid="testid-btn">
                Convert
            </button>
            </LabelInputComponent>
        )
        fireEvent.click(screen.getByTestId('testid-btn'))
        //expect(convertResult).toHaveReturnedWith("tested")
        expect(convertResult).toHaveBeenCalledTimes(1)
        expect(convertResult).toBeDefined();
    })
});
*/