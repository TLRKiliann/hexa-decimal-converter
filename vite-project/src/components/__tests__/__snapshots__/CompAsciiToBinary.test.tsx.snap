// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshat of CompAsciiToBinary 1`] = `
[
  <div
    className="labelinput--box"
  >
    <label>
      ASCII
       :
    </label>
    <input
      data-testid="testid-input"
      onChange={[Function]}
      placeholder="ASCII"
      type="text"
      value=""
    />
  </div>,
  <button
    data-testid="testid-btn"
    onClick={[Function]}
    type="button"
  >
    Convert
  </button>,
  <div
    className="box--result"
  >
    <p>
      Result : 
      Not binary !!!
    </p>
  </div>,
]
`;
